api/v1alpha1/httpchaos_types.go

```diff
@@ -0,0 +1,238 @@
+// Copyright 2020 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// KindHTTPChaos is the kind for http chaos
+const KindHTTPChaos = "HTTPChaos"
+
+func init() {
+	all.register(KindHTTPChaos, &ChaosKind{
+		Chaos:     &HTTPChaos{},
+		ChaosList: &HTTPChaosList{},
+	})
+}
+
+// HTTPAction represents the chaos action about I/O action.
+type HTTPAction string
+
+const (
+	HTTPDelayAction HTTPAction = "delay"
+	HTTPAbortAction            = "abort"
+	HTTPMixedAction            = "mixed"
+)
+
+type Matcher struct {
+	Name           string `json:"name"`
+	ExactMatch     string `json:"exact_match"`
+	RegexMatch     string `json:"regex_match"`
+	SafeRegexMatch string `json:"safe_regex_match"`
+	RangeMatch     string `json:"range_match"`
+	PresentMatch   string `json:"present_match"`
+	PrefixMatch    string `json:"prefix_match"`
+	SuffixMatch    string `json:"suffix_match"`
+	InvertMatch    string `json:"invert_match"`
```

Comment:
Do we need to add `omitempty` tag and make these as pointer?

Reply:
OK,l agree  thanks for the suggestion.

Reply:
I think it was not necessary to make these as pointer ,because these Matches are expected to be short.

Reply:
> I think it was not necessary to make these as pointer ,because these Matches are expected to be short.

I think what @Yisaer means is that we need to seperate the situation that "the user provides an empty string" and "the user doesn't provide anything", and to archieve this target, we can make these pointers.

Reply:
I understand, I will make these pointers, thanks for the suggestion.

Reply:
@longfangsong exactly.